aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -440,24 +440,60 @@ void normal_fill(Tensor& self, const scalar_t mean, const scalar_t std, Generato
   }
 }
 
+std::vector<int64_t> computeStrideForComplex(IntArrayRef oldstride) {
+  auto res = oldstride.vec();
+  for(size_t i = 0; i < res.size(); i++) {
+    res[i] = res[i] * 2;
+  }
+  res.emplace_back(1);
+  return res;
+}
+
+Tensor view_complex_as_float(const Tensor& self) {
+  auto new_sizes = self.sizes().vec();
+  // last dimension will always have two elements containing the real and imag vals
+  new_sizes.emplace_back(2);
+  auto new_strides = computeStrideForComplex(self.strides());
+  if(self.scalar_type() == at::kComplexFloat) {
+    float* data = reinterpret_cast<float*>(self.data_ptr<std::complex<float>>());
+    return at::from_blob(data, new_sizes, ArrayRef<int64_t>(new_strides), dtype(at::kFloat));
+  } else {
+    double* data = reinterpret_cast<double*>(self.data_ptr<std::complex<double>>());
+    return at::from_blob(data, new_sizes, ArrayRef<int64_t>(new_strides), dtype(at::kDouble));
+  }
+  return self;
+}
+
 void normal_kernel(Tensor& self, double mean, double std, Generator* gen) {
-  auto size = self.numel();
-  if (self.scalar_type() == ScalarType::Float && size >= 16 && self.is_contiguous()) {
+  auto float_tensor = self;
```

Comment:
I think I'd suggest just recursively calling `normal_kernel` in the complex case, like `return normal_kernel(view_complex_as_float(self), mean, std/2, gen)`. There are multiple benefits:

1. The code is shorter
2. You don't have to add these random temporary variables (and rename the rest of the function--NB: in the future, rename the parameter, rather than the uses, that reduces the amount of code change!)
3. You avoid an unnecessary refcount bump for input tensors (since `float_tensor` deduces to `Tensor` type, so `Tensor = self` results in a refcount bump)

Reply:
good point! will do